### ISSUE TITLE
Impossible to identify the canvas from C code

### ISSUE THREAD

**Issue by Ecco** [emscripten-core/emscripten#10775]
When using DISABLE_DEPRECATED_FIND_EVENT_TARGET_BEHAVIOR and the new target finding mechanism, it's virtually impossible to identify the canvas from C code.

Indeed, there's a fundamental mismatch between:
* EGL (for instance) expecting `Module["canvas"]` to be a DOM element
* html5.js apis taking selectors to identify an element

So for instance, how would you call `emscripten_set_touchcancel_callback` on the module's canvas?

Maybe #10659 would fix this? Anyway, the current state of code just doesn't work :)

cc @juj

**Comment by juj** [comment]
Indeed, there is an issue here: the problem is that EGL API should be updated to remove the old notion of `Module['canvas']` that should not exist anymore. But EGL and the other APIs do not quite have a maintainer, and are generally not recommended anyways due to other problems (lack of multithreading and OffscreenCanvas support being the main one). I'd rather deprecate EGL, SDL, GLUT and GLFW libraries altogether. (at least until they are rewritten to modern practices)

The proper way to fix this for EGL has been discussed before on this issue tracker: EGL should follow its native window concept to refer to canvases. For example here is how it is referred to on Android: https://developer.android.com/ndk/reference/group/native-activity#anativewindow_fromsurface . Similar to that, there should be an Emscripten EGL specific function that acquires a native window handle from a CSS selector string (well, the CSS selector string itself could be that handle), and EGL should operate on that, instead of relying on magic singleton state `Module['canvas']`.

That would allow creating multiple GL contexts on a web page if there are several canvases, in a way that does not add to code size to those people that are not interested in multiple canvases.

**Comment by Daft-Freak** [comment]
This reminds me of when I was messing with multi-window in SDL2...  I have some EGL stuff here https://github.com/Daft-Freak/emscripten/commits/egl-rewrite from that.

**Comment by Ecco** [comment]
Awesome, another stalebot… This issue is still there…
